firebase-perf/src/main/java/com/google/firebase/perf/metrics/Trace.java

```diff
@@ -630,6 +630,11 @@ private void checkAttribute(@NonNull String key, @NonNull String value) {
       throw new IllegalArgumentException(
           String.format(Locale.ENGLISH, "Trace '%s' has been stopped", name));
     }
+
+    if (key == null || value == null) {
```

Comment:
Is this redundant? We are already validating the key and value in https://github.com/firebase/firebase-android-sdk/blob/master/firebase-perf/src/main/java/com/google/firebase/perf/metrics/Trace.java#L641

Reply:
Yes, this is indeed redundant. Removed this.